Reflection/Reflection/ReflectionTests/BottleReflectionTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Reflection;
+
+namespace ReflectionTests
+{
+    [TestClass]
+    public class BottleReflectionTests
+    {
+        [TestMethod]
+        public void Bottle_Can_Be_Constructed()
+        {
+            Bottle bottle = Activator.CreateInstance(typeof(Bottle)) as Bottle;
+            Assert.IsInstanceOfType(bottle, typeof(Bottle));
+        }
+
+        [TestMethod]
+        public void Bottle_Can_Write_IsInked()
+        {
+            var propertyInfo = typeof(Bottle).GetProperty("IsFull");
```

Comment:
If you change IsFull to private, you can get the property information by adding binding flags, like so:

`var propertyInfo = typeof(Bottle).GetProperty("IsFull", BindingFlags.NonPublic | BindingFlags.Instance);`